src/ashley/auth/backend.py

```diff
@@ -0,0 +1,68 @@
+"""Authentication backend for Ashley"""
+
+import logging
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import PermissionDenied
+
+from ashley.models import AbstractAshleyUser
```

Comment:
Maybe go one step further by using [get_user_model](https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#django.contrib.auth.get_user_model) everywhere you use the user and move the non abstract user model declaration to your sandbox so that the user in the Ashley application can really be overriden.

Reply:
This import is only used for type hinting in this file (and maybe it's useless here ?).
I use `get_user_model()` everywhere in the code.

I'll move the non-abstract user model in the sandbox.

Reply:
Ok I moved the concrete user implementation in the sandbox.
I had to create an app (`ashley_user`) to be able to set `AUTH_USER_MODEL` in the settings.